modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/LogArea.scala

```diff
@@ -82,66 +109,155 @@ object LogArea {
     val reverseLog: FixedLengthBuffer[ServerLogMessage] = log().log.reverse
 
     // Filter according to the levels on the controls
-    private def levelFilter(s: State)(m: ServerLogMessage): Boolean = s.allowedLevel(m.level)
+    private def levelFilter(s: State)(m: ServerLogMessage): Boolean =
+      s.allowedLevel(m.level)
 
-    def rowGetter(s: State)(i: Int): LogRow = reverseLog.filter_(levelFilter(s) _).lift(i).map { l =>
-        val localTime = LocalDateTime.ofInstant(l.timestamp, site.timezone)
-        LogRow(formatter.format(localTime), l.timestamp, l.level, l.msg)
-      }.getOrElse(LogRow.Zero)
+    def rowGetter(s: State)(i: Int): LogRow =
+      reverseLog
+        .filter_(levelFilter(s) _)
+        .lift(i)
+        .map { l =>
+          val localTime = LocalDateTime.ofInstant(l.timestamp, site.timezone)
+          LogRow(formatter.format(localTime), l.timestamp, l.level, l.msg)
+        }
+        .getOrElse(LogRow.Zero)
```

Comment:
If `LogRow` is a monoid you can use `foldMap` here. Otherwise I think `fold(z)(f)` expresses intent a little better than `.map(f).getOrElse(z)`.

Reply:
`LogRow` doesn't make sense as a `Monoid` but will do the `fold`